src/compile.js

```diff
@@ -2051,8 +2051,8 @@ Compiler.prototype.buildcodeobj = function (n, coname, decorator_list, args, cal
     this.u.varDeclsCode += "}";
 
     // inject __class__ cell when running python3
-    if (Sk.python3 && class_for_super) {
-        this.u.varDeclsCode += "$gbl.__class__=$gbl." + class_for_super.v + ";";
+    if (Sk.__future__.python3 && class_for_super) {
+        this.u.varDeclsCode += "$gbl.__class__=this." + class_for_super.v + ";";
     }
```

Comment:
`this` should be changed back go `$gbl` here for this pr.

Reply:
Could you push this change? 😄

Reply:
Done - I had no idea I could push changes to someone else branch like that!